File: UVa/900 - Brick Wall Patterns/henriqueferrolho.cpp
#include <iostream>
using namespace std;

int f(int x) {
	int last = 0;
	int n = 1;

	for (int i = 0; i < x; i++) {
		int next = last + n;
		last = n;
		n = next;
	}

	return n;
}

int main() {
	int l;

	do {
		cin >> l;

		if (l > 0)
			cout << f(l) << endl;
	} while (l != 0);

	return 0;
}

File: UVa/00495 - Fibonacci Freeze/henriqueferrolho.java
import java.io.IOException;
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.Scanner;

public class Main {

	public static ArrayList<BigInteger> fibs = new ArrayList<BigInteger>(5001);

	public static void f() {
		BigInteger last = new BigInteger("0");
		BigInteger n = new BigInteger("1");

		for (int i = 1; i <= 5001; i++) {
			fibs.add(last);

			BigInteger next = last.add(n);
			last = n;
			n = next;
		}
	}

	public static void main(String[] args) throws IOException {
		f();

		Scanner in = new Scanner(System.in);

		do {
			int l = in.nextInt();
			System.out.println("The Fibonacci number for " + l + " is "
					+ fibs.get(l));
		} while (in.hasNextInt());

		in.close();
	}

}
